src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/PlatformNotSupported.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+
+namespace System.Runtime.InteropServices.JavaScript
+{
+    // TODO more after https://github.com/dotnet/runtime/issues/70133
+    [SupportedOSPlatform("browser")]
+    public class JSObject : IDisposable
+    {
+        internal JSObject() => throw new PlatformNotSupportedException();
+        public bool IsDisposed { get => throw new PlatformNotSupportedException(); }
+        public void Dispose() => throw new PlatformNotSupportedException();
+    }
+    [SupportedOSPlatform("browser")]
+    public sealed class JSException : Exception
```

Comment:
Do we really need to platform-lock JSException? Won't that require people to potentially wrap error handlers in ifdefs when having it there won't cause any problems?

Reply:
will put your comment into the code so that we could consider changing it next time we will enter API review.